Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -38,32 +37,16 @@ public class ByteBuffer {
 
     public func putByte(_ value: UInt8) {
         array.append(value)
-        pointer.advanced(by: currentIndex).initialize(to: value)
     }
 
     public func put(_ value: UInt8) -> ByteBuffer {
         array.append(value)
-        pointer.advanced(by: currentIndex).initialize(to: value)
         return self
     }
 
-    public func put(_ value: Data) -> ByteBuffer {
+    public func put(_ value: Data) {
```

Comment:
So just out of curiosity, why does only this overload not return self?

Reply:
idk why I made any of them return self to be honest. I now want to just undo that on all of them. I can't remember if there was a good reason to do so. but this works so I figure ill just confirm this doesnt have any issues moving forward and I can do a separate PR to fix all the others.